### PR TITLE
Bigquery fixes

### PR DESCRIPTION
**Pull request by claudevdm**
Fix the following BigQuery enrichment transform issues
1. Conditions on more than one field returns duplicate rows
2. Batching returns incorrect results (checks equality of restriction field on all columns of the returned data)
3. Batched_condition_template does not wrap conditions in parenthesis

Issues 1 and 2 are fixed by keying batched requests by a tuple of values.
Issue 3 is fixed by wrapping conditions in parenthesis. 

Also modify integration tests to 
1. Set up and tear down a new table on each test
2. Add test cases for above mentioned issues
3. Verify actual enriched pcollection output
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by claudevdm** [comment]
R: @damccorm

File: sdks/python/apache_beam/transforms/enrichment_handlers/bigquery.py [hunk 1]
```diff
@@ -194,14 +202,16 @@ def __call__(self, request: Union[beam.Row, List[beam.Row]], *args, **kwargs):
               "Make sure the values passed in `fields` are the "
               "keys in the input `beam.Row`." + str(e))
         values.extend(current_values)
-        requests_map.update((val, req) for val in current_values)
+        requests_map.update(
+            (self.create_row_key(req), req) for val in current_values)
```
Comment (damccorm) [inline sdks/python/apache_beam/transforms/enrichment_handlers/bigquery.py:206]: Is this now just inserting `(self.create_row_key(req), req)` into the map repeatedly? It doesn't seem like we need the `for val in current_values` anymore

**Review by damccorm — COMMENTED** [review]
Thanks! Just had one comment, otherwise this LGTM

**Review by damccorm — APPROVED** [review]
LGTM, thanks! Yaml test failures are unrelated and failing at head, so merging

**Pull request merged**